nav2_lifecycle/include/nav2_lifecycle/lifecycle_node.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_LIFECYCLE__LIFECYCLE_NODE_HPP_
+#define NAV2_LIFECYCLE__LIFECYCLE_NODE_HPP_
+
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "nav2_lifecycle/lifecycle.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_lifecycle
+{
+
+class LifecycleNode : public rclcpp_lifecycle::LifecycleNode, public ILifecycle
+{
+public:
+  LifecycleNode(
+    const std::string & node_name,
+    const std::string & namespace_ = "",
+    bool use_rclcpp_node = false,
+    const rclcpp::NodeOptions & options = rclcpp::NodeOptions());
+  virtual ~LifecycleNode();
+
+protected:
+  // These two are recommended to be implemented by derived classes but have empty,
+  // default implementations
+  virtual CallbackReturn onShutdown(const rclcpp_lifecycle::State & state);
```

Comment:
This is a bit confusing, having two different methods in the same class that just differ by naming style. Seems like it would be better if this included either `ILifecycle` or `LifecycleNode` using composition rather than inheritance.

Reply:
I'll fix the naming convention to match rclcpp_lifecycle::LifecycleNode and clean this up.